### ISSUE TITLE
stack path --docker-auto-pull prints docker output to stdout, making the result hard to parse

### ISSUE THREAD

**Issue by bitonic** [commercialhaskell/stack#2733]
When running `stack path --docker-auto-pull` if the docker image hasn't been pulled the output of `docker pull` is printed to `stdout`, with the actual data of `stack path` printed at the end.

I think it would be more appropriate to print this kind of output to `stderr`, to keep the `stdout` output reliable and amenable to parsing.

**Comment by lwm** [comment]
Haven't tried this one but after a first glance, my thoughts:
  - The output of `docker pull` is not an error though? It is alternative code path executing. 
  - What do you mean by reliable?

**Comment by bitonic** [comment]
stderr is not used for errors only, but for any information which is not an output of the program.

With "reliable format" I mean one which does not change depending on the environment I'm currently in.

**Comment by lwm** [comment]
Thanks for clarification.

Stack tends to do alternate things depending on your system. For example, if running a stack command inside a project or in the global project. I'm not sure the output of Stack was ever optimised to be machine readable :|

Would passing `--no-docker-auto-pull` make your scripting easier?

**Comment by bitonic** [comment]
I definitely think that `stack path` output should be machine readable.

Not using auto docker pull solves the "spurious output in stdout" problem, but adds the "no automatic docker pull" problem ;)

**Comment by lwm** [comment]
>  but adds the "no automatic docker pull" problem ;) 

Yes, it's another issue. 

But it seems like we're done here regarding making `stack path --docker-auto-pull` more reliable for parsing? Just use `--no-docker-auto-pull`.

**Comment by bitonic** [comment]
I'd like to be able to parse the output reliably while passing `--docker-auto-pull`. So "don't use `--docker-auto-pull` is not really a solution.

**Comment by borsboom** [comment]
It may be possible to capture `docker pull`'s  stdout and send it to stderr instead, and if so that'd be great.  Docker does funny things with the terminal sometimes and I've usually found it best to let it have full control, but maybe `docker pull` won't have that problem.

**Comment by bitonic** [comment]
@borsboom , it'd be a matter of augmenting the `Stack.Run` module to allow to pass explicit handles for stdout and stderr, and then specify stderr as stdout for that invocation.

I can implement this if you think it'd be the way to go.

**Comment by borsboom** [comment]
Yup, that sounds good to me.

**Comment by bitonic** [comment]
OK, I'll submit a PR soon.
